src/replay/consensus/core.zig

```diff
@@ -1,235 +1,661 @@
 const Transaction = sig.core.transaction.Transaction;
 
+const AccountReader = sig.accounts_db.AccountReader;
+
 const LedgerReader = sig.ledger.LedgerReader;
 const LedgerResultWriter = sig.ledger.result_writer.LedgerResultWriter;
 
-const SlotHistoryAccessor = sig.consensus.replay_tower.SlotHistoryAccessor;
-const ReplayTower = sig.consensus.replay_tower.ReplayTower;
-const ProgressMap = sig.consensus.progress_map.ProgressMap;
 const ForkChoice = sig.consensus.fork_choice.ForkChoice;
+const GossipVerifiedVoteHash = sig.consensus.vote_listener.GossipVerifiedVoteHash;
+const HeaviestSubtreeForkChoice = sig.consensus.HeaviestSubtreeForkChoice;
 const LatestValidatorVotes = sig.consensus.latest_validator_votes.LatestValidatorVotes;
+const ParsedVote = sig.consensus.vote_listener.vote_parser.ParsedVote;
+const ProgressMap = sig.consensus.progress_map.ProgressMap;
+const ReplayTower = sig.consensus.replay_tower.ReplayTower;
+const SlotHistoryAccessor = sig.consensus.replay_tower.SlotHistoryAccessor;
+const ThresholdConfirmedSlot = sig.consensus.vote_listener.ThresholdConfirmedSlot;
+const VerifiedVote = sig.consensus.vote_listener.VerifiedVote;
+const VoteListener = sig.consensus.vote_listener.VoteListener;
 
 const SlotTracker = sig.replay.trackers.SlotTracker;
 const EpochTracker = sig.replay.trackers.EpochTracker;
-const SlotData = sig.replay.edge_cases.SlotData;
+
+const AncestorDuplicateSlotToRepair = replay.consensus.edge_cases.AncestorDuplicateSlotToRepair;
+const DuplicateConfirmedState = sig.replay.consensus.edge_cases.DuplicateConfirmedState;
+const SlotData = sig.replay.consensus.edge_cases.SlotData;
+const SlotStatus = sig.replay.consensus.edge_cases.SlotStatus;
+
+const ReplayResult = replay.execution.ReplayResult;
+const ProcessResultParams = replay.consensus.process_result.ProcessResultParams;
 
 const collectVoteLockouts = sig.consensus.replay_tower.collectVoteLockouts;
 const isDuplicateSlotConfirmed = sig.consensus.replay_tower.isDuplicateSlotConfirmed;
-const check_slot_agrees_with_cluster = sig.replay.edge_cases.check_slot_agrees_with_cluster;
+const check_slot_agrees_with_cluster =
+    sig.replay.consensus.edge_cases.check_slot_agrees_with_cluster;
 
 const MAX_VOTE_REFRESH_INTERVAL_MILLIS: usize = 5000;
 
-pub const ConsensusParams = struct {
-    allocator: Allocator,
+/// TowerConsensus contains all the state needed for operating the Tower BFT
+/// consensus mechanism in sig.
+pub const TowerConsensus = struct {
     logger: Logger,
-    replay_tower: *ReplayTower,
-    progress_map: *ProgressMap,
-    slot_tracker: *SlotTracker,
-    epoch_tracker: *const EpochTracker,
-    fork_choice: *ForkChoice,
+    my_identity: Pubkey,
+
+    // Core consensus state
+    fork_choice: HeaviestSubtreeForkChoice,
+    replay_tower: ReplayTower,
+    latest_validator_votes: LatestValidatorVotes,
+    status_cache: sig.core.StatusCache,
+    slot_data: SlotData,
+    arena_state: std.heap.ArenaAllocator.State,
```

Comment:
Should add a comment on this field about the fact that it is used for some temporary allocations that don't outlive functions; ie, it isn't used for any persistent data (unless that has changed during this PR, but I assume that's not the case).

Reply:
c791225b61f076bcf84c6cef8e4e1091211be0ad